### DISCUSSION TITLE
How should we surface chain-specific errors from RPC method calls.

### DISCUSSION THREAD

**Discussion by bugbytesinc** [WalletConnect/walletconnect-monorepo#3789] [Developer Support]
The Hedera community is working on an Hedera Improvement Proposal (HIP) in preparation for eventually creating a pull request in wallet connect for adding some hedera specific network RPC methods.  We think there may be a few hedera chain specific error conditions that a controller might return that are not covered in the WC specs.  Is there a number range that can be used for these specific errors?  Or is it considered a better practice to provide a place to return an error code in RPC method schema?  When I look at the "network_" specific namespaces, I don't see any examples to copy.

For a concrete reference, the Hedera Improvement Proposal (HIP) discussion is here: https://github.com/hashgraph/hedera-improvement-proposal/discussions/819#discussioncomment-7295448

**Comment by bumblefudge** [comment]
Interesting!  Bumble here from CASA.  One idea about how/where to extend the list of error messages would be to create a Hedera<>CAIP-25 profile for inclusion here:
https://namespaces.chainagnostic.org/
No one has yet extended the CAIP-25 spec for namespace-specific error messages or debugging issues, but if I'm being honest, I would warmly welcome the precedent being set!  That profile can be as prescriptive or non-normative/suggestive as you see fit, including being "tentative until contradicted by future HIPs" if that makes sense.  

Feel free to ping me upstream in HIP-land if you'd like help designing anything, otherwise I'll just wait for the PR on CASA repos 😄

**Reply by bugbytesinc** [reply]
Sorry, I'm a bit of a n00b here, if I understand correctly CAIP-25 covers the handshake protocol?  I don't think there are any specific error codes in that context, Hedera has defined CAIP profiles 2, 10, 19 and I think all that conforms with existing session management out-of-the-box.

The edge cases for Hedera concern classes of errors returned from a RPC call, particularly those cases where the controller cannot find or communicate with the target hedera node (all other cases are covered by getting a return value from the hedera node), we could invent method-specific codes to augment the normal hedera "precheck" codes, but thought that it would be more appropriate to return these errors along the same path that `userRejectedRequest` would be returned for example.

**Reply by bugbytesinc** [reply]
(apologies, its possible i may be misusing the vocabulary here, I'm trying fuse two different worlds)

**Reply by bugbytesinc** [reply]
Is there a good reference CAIP-25 example to look at to get an idea how to structure it (even if it does not include custom error codes)?

**Reply by bumblefudge** [reply]
Mmm, sorry, no examples come to mind of a protocol-CAIP getting profiled to have additional error codes, like I said it's a precedent I am eager to see set :D

Maybe this would help? I would say to just compare, the CAIP-10 profiles on namespaces.chainagnostic.org to the original CAIP-10 for an idea of how original-spec and profiles of that spec interact between the two repos.
https://github.com/ChainAgnostic/namespaces/blob/main/CONTRIBUTING.md
